### PR TITLE
[FIX] website_twitter: update the Twitter API guide

### PR DESCRIPTION
**Pull request by Guillaume-gdi**
The way to have API Key and API Secret has changed to configure a
Twitter application. The guide for this configuration is updated with
this commit.

task-2852404
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_twitter/views/res_config_settings_views.xml [hunk 1]
```diff
@@ -30,21 +30,17 @@
                                 <blockquote class="small">
                                     <h2>How to configure the Twitter API access</h2>
                                     <ol>
-                                        <li>Create a new Twitter application on <a href="https://apps.twitter.com/app/new" target="new">https://apps.twitter.com/app/new</a>
-                                              with the following values:
+                                        <li>Log in or create an account on <a href="https://developer.twitter.com/" target="new"> https://developer.twitter.com/</a></li>
+                                        <li>Once connected, go to <a href="https://developer.twitter.com/portal/" target="new">https://developer.twitter.com/portal/</a></li>
+                                        <li>Complete the Twitter portal access process and <strong>create a project</strong> with the following information:
```
Comment (rdeodoo) [inline addons/website_twitter/views/res_config_settings_views.xml:35]: > Complete the Twitter portal access process

I guess you only do that once? I don't see it in my account.

Maybe you could then rephrase it to something like "If not yet done, .." and split the rest in another step (the part about creating a project)
Because from the portal URL you gave, you can't create a project, it is at https://developer.twitter.com/en/portal/projects-and-apps

```suggestion
                                        <li>Once connected and if not already done, complete the Twitter portal access process on <a href="https://developer.twitter.com/portal/" target="new">https://developer.twitter.com/portal/</a></li>
                                        <li><strong>Create a project</strong> on <a href="https://developer.twitter.com/en/portal/projects-and-apps" target="new">https://developer.twitter.com/en/portal/projects-and-apps</a> with the following information:
```

File: addons/website_twitter/views/res_config_settings_views.xml [hunk 2]
```diff
@@ -30,21 +30,17 @@
...
                                             <ul>
                                                 <li><strong>Name: </strong> Odoo Twitter Integration</li>
+                                                <li><strong>Use Case: </strong> Embedding Tweets in a website</li>
                                                 <li><strong>Description: </strong> Odoo Twitter Integration </li>
-                                                <li><strong>Website: </strong>
-                                                <b><field name="twitter_server_uri" /></b></li>
-                                                <li><strong>Callback URL: </strong>Leave blank</li>
-                                                <li>Accept terms of use and click on the Create your Twitter application button at the bottom</li>
+                                                <li><strong>App Name: </strong> choose a unique name </li>
```
Comment (rdeodoo) [inline addons/website_twitter/views/res_config_settings_views.xml:40]:
```suggestion
                                                <li><strong>App Name: </strong> choose an unique name </li>
```

File: addons/website_twitter/views/res_config_settings_views.xml [hunk 3]
```diff
@@ -30,21 +30,17 @@
...
                                     <h2>How to configure the Twitter API access</h2>
                                     <ol>
-                                        <li>Create a new Twitter application on <a href="https://apps.twitter.com/app/new" target="new">https://apps.twitter.com/app/new</a>
-                                              with the following values:
+                                        <li>Log in or create an account on <a href="https://developer.twitter.com/" target="new"> https://developer.twitter.com/</a></li>
+                                        <li>Once connected, go to <a href="https://developer.twitter.com/portal/" target="new">https://developer.twitter.com/portal/</a></li>
+                                        <li>Complete the Twitter portal access process and <strong>create a project</strong> with the following information:
                                             <ul>
                                                 <li><strong>Name: </strong> Odoo Twitter Integration</li>
+                                                <li><strong>Use Case: </strong> Embedding Tweets in a website</li>
```
Comment (rdeodoo) [inline addons/website_twitter/views/res_config_settings_views.xml:38]: Are you sure about all this? The only field I can see is the app name (the first step is just about choosing Dev/Staging/prod)
image

File: addons/website_twitter/views/res_config_settings_views.xml [hunk 4]
```diff
@@ -30,21 +30,17 @@
...
-                                                <li><strong>Callback URL: </strong>Leave blank</li>
-                                                <li>Accept terms of use and click on the Create your Twitter application button at the bottom</li>
+                                                <li><strong>App Name: </strong> choose a unique name </li>
                                             </ul>
                                         </li>
-                                        <li>Switch to the API Keys tab: <br/>
-                                            <img src='/website_twitter/static/src/img/api_key.png' alt="API Key"/>
-                                        </li><br/>
-                                        <li>Copy/Paste Consumer Key (API Key) and Consumer Secret (API Secret) keys below.</li>
+                                        <li>Copy/Paste the API Key and API secret values into the above fields </li>
```
Comment (rdeodoo) [inline addons/website_twitter/views/res_config_settings_views.xml:43]:
```suggestion
                                        <li>Copy/Paste the API Key and API Key Secret values into the above fields </li>
```

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Do you know if the old keys (gotten from the old API URL) are still working and if this change only impacts new apps?

**Comment by Guillaume-gdi** [comment]
> Do you know if the old keys (gotten from the old API URL) are still working and if this change only impacts new apps?

No I can't confirm 100% but I think they still work because the old API still responds but I have the impression that the API v2 keys do not allow calling the old API. Also I don't know what the user was supposed to put as 'twitter_screen_name'..
This is the reason why I put in comment of the task that this should be investigated. I can do that if needed but decisions will have to be made (what to do for users who have already configured the block with the old API, ...). I think that merging this new tutorial while it seems to me that it is impossible for the user to make the block works is not very useful, right?

**Comment by rdeodoo** [comment]
The twitter screen name is just an user name of a twitter user, from which you want to make info appear in the snippet :grimacing: ..

> I think that merging this new tutorial while it seems to me that it is impossible for the user to make the block works is not very useful, right?

What do you mean? The snippet is working perfectly fine? I just created an app and entered my credentials and the screen name of someone and I got the info fetched and displayed :shrug:

File: addons/website_twitter/views/res_config_settings_views.xml (hunk 2 above)
Comment (Guillaume-gdi) [reply addons/website_twitter/views/res_config_settings_views.xml:40]: Mhhh I don't think so... You have to put `an` only before a consonant **sound**. Here the `u` of unique will be pronounced `you` so we leave `a`. Like `a university`

**Comment by Guillaume-gdi** [comment]
Note that the misunderstandings about this PR were due to the fact that to have access to the old Twitter API (v1.1) which is used on Odoo, it was necessary either to have created a project on twitter before the new version of the API was released or to ask to have Elevated access. The guide is updated to specify that you need Elevated access for the block to work on Odoo.

File: addons/website_twitter/views/res_config_settings_views.xml (hunk 2 above)
Comment (rdeodoo) [reply addons/website_twitter/views/res_config_settings_views.xml]: Correct :+1: 
My bad :o

**Review by rdeodoo — APPROVED** [review]
@robodoo r+

IMHO, you could have added the remark about the old accounts in the commit message: any created key/account before some date (we don't know exactly) are not required to ask for Elevated rights and any key will work in Odoo. (we lost quite some time figuring this back then, I would expect anyone investigating this commit later to also loose some time as not mentioned, if he already has an old account).

Also, worth noting that the flow when you create a new account is not exactly the one mentioned here, you create the project before asking for Elevated rights, meaning the keys (that you got first) can't be saved in Odoo until then (as Odoo correctly raise an error about 403).

Anyway, this will already be better as no dead link and correct explanation (even if order is not always right regarding if you already have an account or not).

Plus, it was also tested & approved by PO.
